File: frontend/src/models/member.ts
export interface Member {
  id?: number;
  name: string;
  surname: string;
}

export const buildMember = ({
  id = undefined,
  name = 'default name',
  surname = 'default surname'
}: Partial<Member>): Member => ({
  id,
  name,
  surname
});

File: backend/src/main/java/com/github/wcosmedlr/dto/Member.java
package com.github.wcosmedlr.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder(setterPrefix = "set")
@NoArgsConstructor
public class Member implements Comparable<Member>{

    private Long id;
    private String name;
    private String surname;

    @Override
    public int compareTo(Member o) {
        return this.getName().compareTo(o.getName());
    }

}
